rust/benches/bench.rs

```diff
@@ -0,0 +1,273 @@
+fn serde_json_serialize(b: &mut Bencher) {
+    let log = Log::new();
+    let mut buf = Vec::new();
+
+    serde_json::to_writer(&mut buf, &log).unwrap();
+    b.bytes = buf.len() as u64;
+
+    b.iter(|| {
+        buf.clear();
+        serde_json::to_writer(&mut buf, &log).unwrap();
+    });
+}
+
+#[bench]
+fn serde_json_deserialize(b: &mut Bencher) {
+    let log = Log::new();
+    let json = serde_json::to_string(&log).unwrap();
+
+    b.bytes = json.len() as u64;
+
+    b.iter(|| serde_json::from_str::<Log>(&json).unwrap());
+}
+
+#[bench]
+fn capnp_populate(b: &mut Bencher) {
+    let mut msg = Builder::new_default();
+
+    b.iter(|| {
+        protocol_capnp::populate_log(&mut msg);
+    });
+}
+
+#[bench]
+fn capnp_serialize(b: &mut Bencher) {
+    let mut msg = Builder::new_default();
+    protocol_capnp::populate_log(&mut msg);
+
+    let mut buf = Vec::new();
+    ::capnp::serialize::write_message(&mut buf, &msg).unwrap();
+    b.bytes = buf.len() as u64;
+
+    b.iter(|| {
+        buf.clear();
+        ::capnp::serialize::write_message(&mut buf, &msg).unwrap();
+    });
+}
+
+#[bench]
+fn capnp_serialize_packed(b: &mut Bencher) {
+    let mut msg = Builder::new_default();
+    protocol_capnp::populate_log(&mut msg);
+
+    let mut buf = Vec::new();
+    ::capnp::serialize_packed::write_message(&mut buf, &msg).unwrap();
+    b.bytes = buf.len() as u64;
+
+    b.iter(|| {
+        buf.clear();
+        ::capnp::serialize_packed::write_message(&mut buf, &msg).unwrap();
+    });
+}
+
+#[bench]
+fn capnp_deserialize(b: &mut Bencher) {
+    let mut msg = Builder::new_default();
+    protocol_capnp::populate_log(&mut msg);
+
+    let mut buf = Vec::new();
+    ::capnp::serialize::write_message(&mut buf, &msg).unwrap();
+    b.bytes = buf.len() as u64;
+
+    b.iter(|| {
+        let mut rdr = &*buf;
+        let message_reader =
+            ::capnp::serialize::read_message(&mut rdr, ReaderOptions::new()).unwrap();
+        message_reader.get_root::<log_capnp::log::Reader>().unwrap();
+    });
+}
+
+#[bench]
+fn capnp_deserialize_packed(b: &mut Bencher) {
+    let mut msg = Builder::new_default();
+    protocol_capnp::populate_log(&mut msg);
+
+    let mut buf = Vec::new();
+    ::capnp::serialize_packed::write_message(&mut buf, &msg).unwrap();
+    b.bytes = buf.len() as u64;
+
+    b.iter(|| {
+        let mut rdr = &*buf;
+        let message_reader =
+            ::capnp::serialize_packed::read_message(&mut rdr, ReaderOptions::new()).unwrap();
+        message_reader.get_root::<log_capnp::log::Reader>().unwrap();
+    });
+}
+
+#[bench]
+fn flatbuffers_populate_with_args(b: &mut Bencher) {
+    let mut msg = flatbuffers::FlatBufferBuilder::new();
+
+    b.iter(|| {
+        msg.reset();
+        let log = protocol_flatbuffers::populate_log_with_args(&mut msg);
+        msg.finish(log, None);
+    });
+}
+
+#[bench]
+fn flatbuffers_populate_with_builder(b: &mut Bencher) {
+    let mut msg = flatbuffers::FlatBufferBuilder::new();
```

Comment:
I think this needs to use black_box so that it is not optimized by the compiler. If not the results might look unfair.